doc/iterations/2025.10/add_support_for_indexing_models.adoc

```diff
@@ -0,0 +1,66 @@
+= Add support for indexing models
+
+== Problem
+
+Full-text search (in the command palette) is only considering the label of elements.
+We want to provide a better implementation of full-text search that searches in attributes, and ranks the results.
+
+== Key Result
+
+- The content of projects is indexed in ElasticSearch and up-to-date with the actual project data
+- The way projects are indexed can be configured by downstream applications (a default implementation is provided by Sirius Web)
+- The search command in the command palette relies on ElasticSearch to provide its results (if ElasticSearch is up and running)
+
+=== Acceptance Criteria
+
+- Backend integration tests should also be available to ensure that the core parts of the behavior are working as expected.
+- Performance tests should be done on large models to ensure the solution matches the expectations.
+
+== Solution
+
+Sirius Web will provide a default implementation to index projects:
+
+- Each project will be indexed in a dedicated index
+- Only the semantic data of a project will be indexed
+* The raw Json produced by EMF Json will be used as a baseline to produce documents to index.
+* This baseline can be enriched with additional information computed by the backend (e.g. the label of the element).
+- The search command in the command palette will query ElasticSearch to retrieve objects matching a given term.
+- Results will be presented as before, but their rank in the list will reflect their rank in the query result set.
+
+Indexes will be created on startup (if they do not exist), and kept up-to-date with the changes performed in the indexed models.
+
+=== Scenario
+
+=== Search an element with text from its label
+
+- The user opens the command palette and selects the _Search_ command
+- The user inputs text matching (fully or partially) the label of an element
+- The command palette shows the element in its result
+- The user can click on the result to select the element
+
+=== Search an element with text from an attribute
+
+Same scenario as above, but the input text matches an attribute of an element.
+
+=== Search an element with text matching labels and attributes
+
+Same scenario as above, but matches on labels are presented before matches on attributes in the results.
+
+=== Search an element when ElasticSearch is not running
+
+Sirius Web defaults to the existing search implementation that only looks for text in label.
+
+=== Cutting backs
+
+- (Nice to have) Add a way to discard search results with a confidence below a given threshold.
+
+== Rabbit holes
+
+Fine tuning of the index can be a rabbit hole.
+An initial version of the index based on https://www.elastic.co/docs/manage-data/data-store/mapping/dynamic-mapping[dynamic mappings] will be provided as a basis for future improvements.
+
+== No-gos
+
+- The integration of ES|QL is out of the scope of this shape.
```

Comment:
I restricted the scope of this shape to in-project search to make it more concise.
Let me know if you think these features should be part of this shape.